libmscore/paste.cpp

```diff
@@ -324,9 +324,10 @@ bool Score::pasteStaff(XmlReader& e, Segment* dst, int dstStaff)
 
                               // be sure to paste the element in the destination track;
                               // setting track needs to be repeated, as it might have been overwritten by el->read()
-                              el->setTrack(e.track());
+                              // preserve *voice* from source, though
+                              el->setTrack(e.track() + el->voice());
 
-                              undoChangeElement(seg->element(e.track()), el);
+                              undoAddElement(el);
```

Comment:
I guess another way would be to find the right element that we could eventually "change" by searching the annotations() elements for a matching type... Not sure if it's worth the trouble though.